wal/src/lib.rs

```
@@ -730,7 +736,7 @@ mod tests {
         let mut reader = wal.reader_for_segment(closed.id).unwrap();
 
         let mut ops = vec![];
-        while let Ok(Some(mut batch)) = reader.next_batch() {
+        while let Some(Ok(mut batch)) = reader.next() {
```

Similarly to above, I think this ignores errors, which may not be desirable

---

I think in my haste to update the tests I missed out that some of these `while` loops can now be done by combinators too!